src/notebooksyncagent.cpp

```diff
@@ -750,6 +751,7 @@ bool NotebookSyncAgent::calculateDelta(
     // and createdBefore; so instead we have to build a datetime which "should" satisfy
     // the inequality for all possible local modifications detectable since the last sync.
     KDateTime syncDateTime = mNotebook->syncDate().addSecs(1); // deleted after, created before...
+    KDateTime earliestAnchorTimestamp = syncDateTime < fromDate ? syncDateTime : fromDate;
```

Comment:
You're right.  Will undo the changes to the other ones, and use the earlier timestamp for the call to insertedIncidences here.  Thanks.